idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/settings/SettingsScreen.kt

```diff
@@ -93,64 +84,38 @@ private fun SettingsUI(
-        SectionTitle(
-            name = "Plugin mode: $modeName",
-            paddingValues = PaddingValues(start = 24.dp),
-        )
-        TextButton(
-            modifier = Modifier.padding(start = 12.dp),
-            onClick = onChangeMode,
-        ) {
-            Text(
-                text = "Change mode",
-                style = MaterialTheme.typography.bodyMedium,
-            )
-        }
+        var selectedTabIndex by rememberMutableState { 0 }
+        val tabs = listOf("General", "Export", "Preview", "About")
 
-        SectionTitle(name = "ImageVector export settings")
-        VerticalSpacer(8.dp)
-        OutputFormatSection(
-            outputFormat = settings.outputFormat,
-            onAction = onAction,
-        )
-        VerticalSpacer(8.dp)
-        Row(
+        ScrollableTabRow(
+            tabs = tabs,
+            selectedTabIndex = selectedTabIndex,
+        ) { tabIndex ->
+            selectedTabIndex = tabIndex
+        }
+        Crossfade(
             modifier = Modifier
                 .fillMaxWidth()
-                .height(48.dp)
-                .toggleable(
-                    value = settings.generatePreview,
-                    onValueChange = { onAction(UpdatePreviewGeneration(it)) },
+                .weight(1f),
+            targetState = selectedTabIndex,
+        ) { index ->
+            when (index) {
+                0 -> GeneralSettings(
+                    mode = settings.mode,
+                    onChangeMode = onChangeMode,
+                    onClearSettings = onClearSettings,
                 )
-                .padding(start = 24.dp, end = 24.dp),
-            verticalAlignment = Alignment.CenterVertically,
-        ) {
-            Text(
-                modifier = Modifier.weight(1f),
-                text = "Include @Preview",
-                style = MaterialTheme.typography.bodyMedium,
-            )
-            Switch(
-                modifier = Modifier.scale(0.9f),
-                checked = settings.generatePreview,
-                onCheckedChange = { onAction(UpdatePreviewGeneration(it)) },
-            )
-        }
-        SectionTitle(name = "Danger zone")
-        TextButton(
-            modifier = Modifier.padding(horizontal = 16.dp),
-            colors = ButtonDefaults.textButtonColors().copy(
-                contentColor = MaterialTheme.colorScheme.error,
-            ),
-            onClick = onClearSettings,
-        ) {
-            Text(text = "Clear all plugin settings")
+                1 -> ImageVectorExportSettings(
+                    onAction = onAction,
+                    outputFormat = settings.outputFormat,
+                    generatePreview = settings.generatePreview,
+                )
+                2 -> ImageVectorPreviewSettings(
+                    showImageVectorPreview = settings.showImageVectorPreview,
+                    onAction = onAction,
+                )
+                3 -> AboutSettings()
+            }
```

Comment:
I guess these could be placed in an enum or type-safe navigation? e.g

```kt
enum class Screen {
    First {
        override fun Content() {
            TODO("Not yet implemented")
        }
    },
    Second {...}
    Third {...}
    Forth {...}
    ;
    @Composable
    abstract fun Content()
}
```

Reply:
can be done with Tiamat and separate tabs will be as a `Screen`.
will take a look 🙂

Reply:
Made all tabs as nested independent screens 🙂